packages/gatsby-image/src/index.js

```diff
@@ -1,18 +1,43 @@
 import React from "react"
 import PropTypes from "prop-types"
 
-// Handle legacy names for image queries.
-const convertProps = props => {
-  let convertedProps = { ...props }
-  if (convertedProps.resolutions) {
-    convertedProps.fixed = convertedProps.resolutions
+const logDeprecationNotice = (prop, replacement) => {
+  if (process.env.NODE_ENV === `production`) {
+    return
+  }
+
+  console.log(
+    `
+    The "${prop}" prop is now deprecated and will be removed in the next major version
+    of "gatsby-image".
+    `
+  )
+
+  if (replacement) {
+    console.log(`Please use ${replacement} instead of "${prop}".`)
+  }
+}
+
+// Handle legacy props during their deprecation phase
+const convertProps = convertedProps => {
+  let { resolutions, sizes, critical } = convertedProps
+
+  // `resolutions`/`sizes` => `fixed`/`fluid`
+  if (resolutions) {
+    convertedProps.fixed = resolutions
     delete convertedProps.resolutions
   }
-  if (convertedProps.sizes) {
-    convertedProps.fluid = convertedProps.sizes
+  if (sizes) {
+    convertedProps.fluid = sizes
     delete convertedProps.sizes
   }
 
+  // `critical` => `loading`
+  if (critical) {
+    logDeprecationNotice(`critical`, `the native "loading" attribute`)
```

Comment:
```suggestion
if (process.env.NODE_ENV === `production`) {
    logDeprecationNotice(`critical`, `the native "loading" attribute`)
}
```

This way the logDeprecationNotice function will be removed as it will be marked as unused. By doing it inside the function it will be kept in production mode which is extra bytes we don't really want.

Reply:
The production environment check was moved into the `logDeprecationNotice()` method itself, webpack will realize the function is a no-op and optimize it out along with the function that calls it.

No need to make this change :)

Reply:
Sweet, didn't know that!